packages/java/endpoint/src/main/java/com/vaadin/hilla/route/RouteUnifyingIndexHtmlRequestListener.java

```diff
@@ -95,7 +107,8 @@ public void modifyIndexHtmlResponse(IndexHtmlResponse response) {
         if (exposeServerRoutesToClient) {
             LOGGER.debug(
                     "Exposing server-side views to the client based on user configuration");
-            availableViews.putAll(collectServerViews());
+            availableViews
+                    .putAll(collectServerViews(response.getVaadinRequest()));
```

Comment:
Earlier we added an application property called `vaadin.expose-server-routes-to-client` with a default value of `false` which is checked here to prevent accidental leakage of server routes to the client, and now that the `populateClientSideMenu` property defined in vaadin/flow#19202, this `exposeServerRoutesToClient` check seems to be a duplicate, and none of the changes in this PR will affect anything until that property and this condition is removed from Hilla codebase.

Reply:
Either default can be changed to `true` or `vaadin.expose-server-routes-to-client` property can be removed from Hilla. I'm not sure if it would make sense to still have it (with different name) in Flow side to make up default value for populateClientSideMenu in `DefaultMenuAccessControl`. There's no such system property currently in the other PR.

Reply:
Like discussed in the Flow daily, the decision on whether to send Flow views should be made on the Hilla side. Flow should only make the data available to Hilla in case Hilla wants to use it.

This can very well be in the form of the existing `expose-server-routes-to-client` property (even though we usually have a hierarchy of properties separated by `.` rather than a long phrase separated by `-`). We could just change it so the default in case the property is not explicitly set varies depending on whether Hilla knows that it has a client-side main layout.

Reply:
> the decision on whether to send Flow views should be made on the Hilla side

Could you please elaborate more about the "why"?

Reply:
It's based on a technical practicality: Hilla knows whether there is a client-side main layout but Flow doesn't (without making Flow using Hilla APIs which would lead to a circular dependency).

Reply:
Thanks. We agreed on the same in https://github.com/vaadin/flow/pull/19202#discussion_r1572233777